File: templates/hashmap/hashmap.go
package hashmap

func TwoSum(nums []int, target int) (int, int, bool) {
	index := map[int]int{}
	for i, v := range nums {
		if j, ok := index[target-v]; ok {
			return j, i, true
		}
		index[v] = i
	}
	return -1, -1, false
}

func TopKFrequent(nums []int, k int) []int {
	freq := map[int]int{}
	for _, v := range nums {
		freq[v]++
	}

	buckets := make([][]int, len(nums)+1)
	for v, count := range freq {
		buckets[count] = append(buckets[count], v)
	}

	result := make([]int, 0, k)
	for count := len(buckets) - 1; count >= 0 && len(result) < k; count-- {
		result = append(result, buckets[count]...)
	}
	if len(result) > k {
		result = result[:k]
	}
	return result
}

func LongestConsecutive(nums []int) int {
	seen := map[int]bool{}
	for _, v := range nums {
		seen[v] = true
	}

	best := 0
	for _, v := range nums {
		if seen[v-1] {
			continue
		}
		length := 1
		for seen[v+length] {
			length++
		}
		if length > best {
			best = length
		}
	}
	return best
}

File: templates/hashmap/hashmap.py
def two_sum(nums: list[int], target: int) -> tuple[int, int, bool]:
    index: dict[int, int] = {}
    for i, value in enumerate(nums):
        if target - value in index:
            return index[target - value], i, True
        index[value] = i
    return -1, -1, False


def top_k_frequent(nums: list[int], k: int) -> list[int]:
    freq: dict[int, int] = {}
    for value in nums:
        freq[value] = freq.get(value, 0) + 1

    buckets: list[list[int]] = [[] for _ in range(len(nums) + 1)]
    for value, count in freq.items():
        buckets[count].append(value)

    result: list[int] = []
    for count in range(len(buckets) - 1, -1, -1):
        result.extend(buckets[count])
        if len(result) >= k:
            return result[:k]
    return result


def longest_consecutive(nums: list[int]) -> int:
    seen = set(nums)
    best = 0

    for value in nums:
        if value - 1 in seen:
            continue
        length = 1
        while value + length in seen:
            length += 1
        best = max(best, length)

    return best
